code/game/objects/effects/spawners/random/structure.dm

```diff
@@ -93,9 +93,8 @@
 
 /obj/effect/spawner/random/structure/crate_empty/make_item(spawn_loc, type_path_to_make)
 	var/obj/structure/closet/crate/peek_a_boo = ..()
-	if(istype(peek_a_boo))
-		peek_a_boo.opened = prob(50)
-		peek_a_boo.update_appearance()
+	if(istype(peek_a_boo) && prob(50))
+		peek_a_boo.open(null, special_effects = FALSE) //the crate appears immediatly out of thin air so no need to animate anything
```

Comment:
I added this var special_effects so closets opened during mapload don't play sounds or animate unnecessarily since the round is still loading and there is no one to watch, i'll skip the 1st argument yup